### DISCUSSION TITLE
CPV code list in the Notice editor

### DISCUSSION THREAD

**Discussion by teslavitas** [OP-TED/eForms-SDK#432] [Q&A] [answered]
When I look at the field "Main classification" (BT-262-Lot) for notice type 16 in a local instance of the Notice editor, there are about 50 values in the dropdown.
![image](https://user-images.githubusercontent.com/11631176/234528648-8a2b19f5-8c41-4ea2-aac9-e76c6f6c3bd1.png)

This field uses code list "cpv", which has many thousands of options. I can see all those option in a network request to http://localhost:8080/sdk/1.5/codelists/cpv.gc/lang/en
Could you explain how it is reduced to a limited number of options in the dopdown?
Does it has anything to do with the fact that CPV code list is hierarchical?

**Comment by rouschr** [comment, marked as answer]
This looks like a default limit in the UI.
I did a search in the entire project for the number 50 (...) and found this:
`...,maxOptions:50,...` in the code of tom-select javascript so you could tell tomselect to accept more values, with potential issues in the performance I suppose. This would be the proper solution.
In the editor demo we do not support hierarchical selection yet and probably will not have a priority for this as the goal is not to redo something like enotices2. For demo purposes it is OK as long as you can select a value.

Note that version 1.7 will provide hierarchical data in the .gc file of CPV and NUTS (parentCode).
So if in the UI you want to do a nice two or more step selection you could implement it, but it would require more work.

**Reply by teslavitas** [reply]
Thank you, so a simple but correct solution will be to show all the options. I was afraid that I am missing filtering somewhere.

**Reply by rouschr** [reply]
Yes if performance is not an issue, you could try to load all items to at least get something working.
Ideally your user interface should use the multi-step process (hierarchical), but wait for SDK 1.7 for that.
On the server side you could parse that CPV xml and get data in several calls.
On the UI side you would need to have something special for those hierarchical codelists ... as you can imagine this involves quite a lot of work to get a nice result.
